### PR TITLE
[#4380] Improve browser security warning

### LINKED ISSUE #4380: Improve browser security warning
### User Story

As a user, I want to know about the security of websites in the Status browser.

### Description

*Type*: Feature

*Summary*: 
Design update for #4298:

- For **Status curated DApps and websites with valid `HTTPS` certificate**, show security lock icon in the browser bar. Message on-click:
```Connection is secure. Make sure you really trust this site before signing transactions or entering personal data.```

[Link to icon SVG ](https://status-im.slack.com/files/U61UDMS5D/FBDG257GA/icon_lock-black.svg)

e.g.

<br><Br>

- For **websites with no SSL certificate**, show warning with icon in the browser bar. Message on-click:
```Connection is not secure! Do not sign transactions or send personal data on this site.```

[Link to icon SVG ](https://status-im.slack.com/files/U61UDMS5D/FBEDL2UNA/icon_lock_opened-black.svg)

- For **websites with invalid `HTTPS` certificate**, block users from navigating to these sites and present an error page. 

@jeluard which error page do we need here? 

#### Expected behavior
For HTTPS sites, a security icon in browser bar displays the appropriate security message on-click.

For others, a warning message about the insecure connection is presented. 

#### Actual behavior
A warning message is displayed for any site.

### PR DESCRIPTION
**Pull request by flexsurfer**
fixes #4380

### CODE REVIEW

File: src/status_im/models/browser.cljs [hunk 1]
```diff
@@ -14,8 +16,15 @@
 (defn can-go-forward? [{:keys [history-index history]}]
   (< history-index (dec (count history))))
 
+(defn check-if-dapp-in-list [{:keys [history history-index] :as browser}]
+  (let [history-host  (http/url-host (try (nth history history-index) (catch js/Error _)))
+        dapp (first (filter #(= history-host (http/url-host (:dapp-url %))) (apply concat (mapv :data default-dapps/all))))]
```
Comment (goranjovic) [inline src/status_im/models/browser.cljs:21]: this line is a bit too long, can we have it in a helper fn?

**Review by goranjovic — APPROVED** [review]

**Review by jeluard — APPROVED** [review]

**Comment by Serhy** [comment]
@flexsurfer tested on iOS (11.2.5), Android 6.0.1, 7.0 - and this looks good! Now https websites (and Dapps) have a "lock" icon. 
HTTP connection has unlocked icon. 
Both cases have a respective message notifying about connection is secure/insecure.

Peepeth navigates right to https. And, LivePeer Dapp is https://media.livepeer.org/ from now 👍 

When there is an expired/invalid certificate: Android app browser has an 
`Unable to load page -1 There was a network error`. iOS browser respond with the Unable to load page -1202. The certificate for this server is invalid. You might be connecting to a server that is pretending to be “_website.uri.here_” which could put your confidential information at risk.

screen shot 2018-07-30 at 19 10 13


However, there are instances when we display connection as ‘secure’ for certificates having some issues:
On **Android only**: 
- Revoked certificate: https://revoked.badssl.com/
- When intermediate certificate issuer signed with SHA1: e.g. https://sha1-intermediate.badssl.com/

On both Android and iOS app:
- Bad 'HTTP Public Key Pinning' and 'invalid sign certificate timestamp' (well both are only deprecated in Chrome recently, - FF, Safari, Opera are fine with it): e.g. navigate to https://pinning-test.badssl.com/, https://invalid-expected-sct.badssl.com/ 

I’d say the last item is not extremely important, but it would be great if we: 
1) Have the `The certificate for this server is invalid. You might be connecting to a server that is pretending to be “_website.uri.here_” which could put your confidential information at risk.` message on Android browser for insecure connections (so user would know what happened with webpage/Dapp
2) Review whether we can block on Android (like we have for iOS) https with revoked certificate (https://revoked.badssl.com/ ) and SHA1 signed certificates ( https://sha1-intermediate.badssl.com/)

Or we could look at it in separate issues, it's already a great change here! What do you think, @flexsurfer ?

**Comment by flexsurfer** [comment]
hey @Serhy , unfortunately, we can't manage this because it managed by OS

**Comment by Serhy** [comment]
Ok, got it @flexsurfer. Let's merge then!

**Pull request merged**